### DISCUSSION TITLE
How to uncover TS build errors without building whole repo

### DISCUSSION THREAD

**Discussion by jvskriubakken** [nrwl/nx#7742] [Show and Tell]
Hi,

to ensure I'm not committing any TS build errors I run the same command as the CI locally:
```
nx run-many --target=build --all --prod
```

Running this command takes quite a while... I am using IntelliJ and would expect IntelliJ to uncover these TS build errors as I type, but it does not. 

How are you guys ensuring a faster development, without waiting for Nx to build all TS code in repo?

Thanks!

Jørund

**Comment by brandonroberts** [comment]
Hi,

You shouldn't be building everything in CI. The affected commands will build only what's impacted by a particular change.

```ts
nx affected --target build --prod
```

Docs guide: https://nx.dev/l/a/core-extended/affected

**Comment by jvskriubakken** [comment]
I've changed the CI files to now run only affected. However it seems to complain about the given "master branch name" "main", not existing.

```
yarn run v1.22.17
$ nx affected --base=main --target=test
fatal: Not a valid object name main
fatal: No such ref: 'main'
nx affected

Run task for affected projects

Run command using --base=[SHA1] (affected by the committed, uncommitted and
untracked changes):
  --base  Base of the current branch (usually master)                   [string]

or using --base=[SHA1] --head=[SHA2] (affected by the committed changes):
  --base  Base of the current branch (usually master)                   [string]
  --head  Latest commit of the current branch (usually HEAD)            [string]

```

We are using Azure as build system. Could it be this error occurs because the repo has been checked out in "detached head" state?

**Reply by brandonroberts** [reply]
Probably so. We have a guide to help with Azure Pipelines you can use for reference https://nx.dev/l/a/ci/monorepo-ci-azure
